integr8ly/integreatly-operator
# Intly 3413 - Add dedicated admin to user-sso

### What:
Users added to the dedicated-admin group in OpenShift should be created as admins in the master realm of the usersso Keycloak.
Users deleted from the group should be removed from the master realm.
Users should be able to log in to Keycloak with their OpenShift credentials via openshift v4 Oauth client. 

### Verification Steps
- Install the Operator
- Verify that usersso is installed with a master Realm and default admin user
- run CUSTOM_PWD=Password1 ./scripts/set-htpass-idp.sh
- This will create 2 users, customer-admin and test-user. 
- customer-admin is added to the dedicated-admins group. test-user is not
- Log into OpenShift as customer-admin and test-user. This will create a user and identity record for each. Verify with oc get users and oc get identities
- Log into Keycloak as the admin user.
- Verify that customer-admin exists in the master realm and test-user does not (as it is not in the dedicated-admins group)
- Log out of Keycloak and log back in as customer-admin, using the OpenShift-v4 IDP option
- At this point you will be requested to link the customer-admin accounts. Retrieve the customer-admin password from the Keycloak-User YAML to do so. (There is an [open JIRA to remove this linking](https://issues.redhat.com/browse/INTLY-3957))

**Removing User**
- Remove customer-admin from dedicated-admins group
- oc adm groups remove-users dedicated-admins customer-admin
- Log back into Keycloak as the admin user and verify that the customer-admin has been removed from the master Realm
